src/main/java/de/dennisguse/opentracks/services/TrackRecordingService.java

```diff
@@ -417,17 +364,20 @@ private void startGps() {
         showNotification(true);
     }
 
-    void endCurrentTrack() {
+    Track.Id endCurrentTrack() {
         if (!isRecording()) {
             Log.d(TAG, "Ignore endCurrentTrack. Not recording.");
-            return;
+            return null;
         }
 
-        // Need to remember the recordingTrackId before setting it to -1L
+        Track.Id endTrackId = serviceStatus.getRecordingTrackId();
```

Comment:
endTrackId.equals(trackId), right?
So, we can return trackId, right?

Reply:
Yes, we can.

What a catch!!!! I didn't realize :+1: